tests/agemoea_test.cpp

```diff
@@ -0,0 +1,332 @@
+        break;
+      }
+    }
+
+    if (allInRange)
+    {
+      success = true;
+      break;
+    }
+  }
+
+  REQUIRE(success == true);
+}
+
+/**
+ * Optimize for the Schaffer N.1 function using AGE-MOEA optimizer.
+ * Tests for data of type double.
+ */
+TEST_CASE("AGEMOEASchafferN1TestVectorDoubleBounds", "[AGEMOEATest]")
+{
+  // This test can be a little flaky, so we try it a few times.
+  SchafferFunctionN1<arma::mat> SCH;
+  const arma::vec lowerBound = {-1000};
+  const arma::vec upperBound = {1000};
+  const double expectedLowerBound = 0.0;
+  const double expectedUpperBound = 2.0;
+
+  AGEMOEA opt(20, 300, 0.6, 20, 1e-6, 20, lowerBound, upperBound);
+
+  typedef decltype(SCH.objectiveA) ObjectiveTypeA;
+  typedef decltype(SCH.objectiveB) ObjectiveTypeB;
+
+  bool success = false;
+  for (size_t trial = 0; trial < 3; ++trial)
+  {
+    arma::mat coords = SCH.GetInitialPoint();
+    std::tuple<ObjectiveTypeA, ObjectiveTypeB> objectives = SCH.GetObjectives();
+
+    opt.Optimize(objectives, coords);
+    arma::cube paretoSet = opt.ParetoSet();
+
+    bool allInRange = true;
+
+    for (size_t solutionIdx = 0; solutionIdx < paretoSet.n_slices; ++solutionIdx)
+    {
+      double val = arma::as_scalar(paretoSet.slice(solutionIdx));
+      if (!IsInBounds<double>(val, expectedLowerBound, expectedUpperBound, 0.1))
+      {
+        allInRange = false;
+        break;
+      }
+    }
+
+    if (allInRange)
+    {
+      success = true;
+      break;
+    }
+  }
+
+  REQUIRE(success == true);
+}
+
+/**
+ * Optimize for the Fonseca Fleming function using AGE-MOEA optimizer.
+ * Tests for data of type double.
+ */
+TEST_CASE("AGEMOEAFonsecaFlemingDoubleTest", "[AGEMOEATest]")
+{
+  FonsecaFlemingFunction<arma::mat> FON;
+  const double lowerBound = -4;
+  const double upperBound = 4;
+  const double expectedLowerBound = -1.0 / sqrt(3);
+  const double expectedUpperBound = 1.0 / sqrt(3);
+
+  AGEMOEA opt(20, 300, 0.6, 20, 1e-6, 20, lowerBound, upperBound);
+  typedef decltype(FON.objectiveA) ObjectiveTypeA;
+  typedef decltype(FON.objectiveB) ObjectiveTypeB;
+
+  bool success = false;
+  for (size_t trial = 0; trial < 3; ++trial)
+  {
+    arma::mat coords = FON.GetInitialPoint();
+    std::tuple<ObjectiveTypeA, ObjectiveTypeB> objectives = FON.GetObjectives();
+
+    opt.Optimize(objectives, coords);
+    arma::cube paretoSet = opt.ParetoSet();
+
+    bool allInRange = true;
+
+    for (size_t solutionIdx = 0; solutionIdx < paretoSet.n_slices; ++solutionIdx)
+    {
+      const arma::mat solution = paretoSet.slice(solutionIdx);
+      double valX = arma::as_scalar(solution(0));
+      double valY = arma::as_scalar(solution(1));
+      double valZ = arma::as_scalar(solution(2));
+
+      if (!IsInBounds<double>(valX, expectedLowerBound, expectedUpperBound, 0.3) ||
+          !IsInBounds<double>(valY, expectedLowerBound, expectedUpperBound, 0.3) ||
+         !IsInBounds<double>(valZ, expectedLowerBound, expectedUpperBound, 0.3))
```

Comment:
```suggestion
          !IsInBounds<double>(valZ, expectedLowerBound, expectedUpperBound, 0.3))
```

to align with the other parameters.